cedar-policy/src/api.rs

```diff
@@ -111,18 +128,52 @@ impl Entity {
     }
 
     /// Create a new `Entity` with this Uid, no attributes, and no parents.
+    /// ```
+    /// use cedar_policy::{Entity, EntityId, EntityTypeName, EntityUid};
+    /// # use std::str::FromStr;
+    /// let eid = EntityId::from_str("alice").unwrap();
+    /// let type_name = EntityTypeName::from_str("User").unwrap();
+    /// let euid = EntityUid::from_type_name_and_id(type_name, eid);
+    /// let alice = Entity::with_uid(euid);
+    /// # assert_eq!(alice.attr("age"), None);
+    /// ```
     pub fn with_uid(uid: EntityUid) -> Self {
         Self(ast::Entity::with_uid(uid.0))
     }
 
     /// Get the Uid of this entity
+    /// ```
+    /// # use cedar_policy::{Entity, EntityId, EntityTypeName, EntityUid};
+    /// # use std::str::FromStr;
+    /// # let eid = EntityId::from_str("alice").unwrap();
+    /// let type_name = EntityTypeName::from_str("User").unwrap();
+    /// let euid = EntityUid::from_type_name_and_id(type_name, eid);
+    /// let alice = Entity::with_uid(euid.clone());
+    /// assert_eq!(alice.uid(), euid);
+    /// ```
     pub fn uid(&self) -> EntityUid {
         EntityUid(self.0.uid())
     }
 
     /// Get the value for the given attribute, or `None` if not present.
     ///
     /// This can also return Some(Err) if the attribute had an illegal value.
+    /// ```
+    /// use cedar_policy::{Entity, EntityId, EntityTypeName, EntityUid, EvalResult,
+    ///     RestrictedExpression};
+    /// use std::collections::{HashMap, HashSet};
+    /// use std::str::FromStr;
+    /// let eid = EntityId::from_str("alice").unwrap();
+    /// let type_name = EntityTypeName::from_str("User").unwrap();
+    /// let euid = EntityUid::from_type_name_and_id(type_name, eid);
```

Comment:
For this snippet, let's hide the imports, the `let eid` and `let type_name`, and maybe even the `let euid`
```suggestion
    /// # use cedar_policy::{Entity, EntityId, EntityTypeName, EntityUid, EvalResult,
    ///     RestrictedExpression};
    /// # use std::collections::{HashMap, HashSet};
    /// # use std::str::FromStr;
    /// # let eid = EntityId::from_str("alice").unwrap();
    /// # let type_name = EntityTypeName::from_str("User").unwrap();
    /// # let euid = EntityUid::from_type_name_and_id(type_name, eid);
```